src/resumable-upload.ts

```diff
@@ -260,20 +260,23 @@ export class Upload extends Writable {
     uri: uuid.v4(),
     offset: uuid.v4(),
   };
-  private upstreamChunkBuffer: Buffer = Buffer.alloc(0);
-  private chunkBufferEncoding?: BufferEncoding = undefined;
+  /**
+   * A cache of buffers written to this instance, ready for consuming
+   */
+  private writeBuffers: Buffer[] = [];
+  private bufferEncoding?: BufferEncoding = undefined;
```

Comment:
I think you mentioned removing this in our discussion but does encoding actually ever get used for anything?

Reply:
Not really; only for incoming buffers that are encoded to strings - which we decode here:

https://github.com/googleapis/nodejs-storage/blob/efb99b8aad6dbb8cf698cfcabc6a4d1fc3bbefd8/src/resumable-upload.ts#L398